### PR TITLE
Send idle bombers to newly built airpad

### LINKED ISSUE #3979: Feature; unloaded planes use newly built pads
When waitting for a spot on a rearming pad planes will react to pads being built and use them for rearming.

### PR DESCRIPTION
**Pull request by Legomenon-gh**
Fixes #3979, looking for feedback on Lua/Spring library, and if there are major flaws with my logic (the bomber logic is pretty complicated so I suspected I missed something somewhere, especially around queued commands).

I've verified this behaves fine in normal 1v1 games but cannot vouch for edge cases in a large teamgame.

### CODE REVIEW

File: LuaRules/Gadgets/unit_refuel_pad_handler.lua [hunk 1]
```diff
@@ -588,3 +588,45 @@ function gadget:GameFrame(f)
...
+	else
+		bombersToRearm = GetFirstNearestIdleBombers(unitID, idleBombers, padCount)
+	end
+
+	GG.RedirectBombersToAirpad(bombersToRearm, unitID)
+end
+
+function GetFirstNearestIdleBombers(airpadID, idleBombers, count)
+	local unitIDsByDistance = {}
+	for i = 1, #idleBombers do
```
Comment (Legomenon-gh) [inline LuaRules/Gadgets/unit_refuel_pad_handler.lua:613]: There has to be a simpler way to do this... all I want is to sort `idleBombers` by their distance and return the first X bombers, but Lua tables really don't do ordering (or mappings)

**Comment by Legomenon-gh** [comment]
I could throw in https://zero-k.info/Forum/Thread/31656 too (disable AA from hitting grounded targets) if this PR looks sane and nobody else wants to work on that since I'm in that code already

**Comment by GoogleFrog** [comment]
Don't do them both in the same PR. They should be handled distinctly.

**Comment by Legomenon-gh** [comment]
Of course that would be a separate PR. I'm not going to touch the AA stuff though since that's out of my Spring-fu zone to do well

**Comment by Legomenon-gh** [comment]
Closing this PR since I have insufficient confidence in its validity and I believe it can be implemented better

**Pull request closed without merging**